### PR TITLE
[#443] enable customization of target with specVersion

### LINKED ISSUE #443: SpecVersion defined to 2.2 in maven-plugin
Old parameter SpecVersion still defined to 2.2 in MavenPlugin Mojo
Need to upgrade to 3.0 by default and see if possible to set to 2.3 and generate `javax` code from XJC 4

### PR DESCRIPTION
**Pull request by laurentschoelens**
Fixes #443 
To be able to define target from specVersion mojo property and fix default value

### CODE REVIEW

File: maven-plugin/tests/po-2.3/pom.xml [hunk 1]
```diff
@@ -12,13 +12,22 @@
     <dependency>
       <groupId>org.glassfish.jaxb</groupId>
       <artifactId>jaxb-runtime</artifactId>
+      <version>2.3.8</version>
```
Comment (mattrpav) [inline maven-plugin/tests/po-2.3/pom.xml:15]: Can we show users how to use the jaxb-bom v2 here?

**Review by mattrpav — CHANGES_REQUESTED** [review]
This could get tricky to maintain javax and jakarta across the same tooling. Also, this can get confusing to users b/c we have multi-branch releases right now.

I think as long as the testing and upstream jaxb support it properly (api v impl separation) than we can support it.

I don't want to do crazy gymnastics to support cross versions, when we can just support releases across branches.

**Comment by mattrpav** [comment]
The more I think about this, I'm not sure we want to support it across _major_ versions. 

We have jaxb-plugins, jaxb-plugins-runtime, jaxb-plugin-annotate, all that have hard dependency on javax v jakarta. I think it would be weird for a project to use this across all the things. The dep tree would look like:

jaxb-maven-plugin v4.x (specVersion = 2)
jaxb-plugins v2.x
jaxb-plugins-runtime v2.x
jaxb-plugin-annotate v2.x

What does support across major versions get us or help users?

**Comment by laurentschoelens** [comment]
> The more I think about this, I'm not sure we want to support it across _major_ versions of jaxb. For sure, this is super great for _minor_ version jaxb spec bumps. (ie jaxb 4.0, jaxb 4.1 etc)
> 
> We have jaxb-plugins, jaxb-plugins-runtime, jaxb-plugin-annotate, all that have hard dependency on javax v jakarta. I think it would be weird for a project to use this across all the things. The dep tree would look like:
> 
> jaxb-maven-plugin v4.x (specVersion = 2) jaxb-plugins v2.x jaxb-plugins-runtime v2.x jaxb-plugin-annotate v2.x
> 
> What does support across major versions get us or help users?

As discussed : the major problem would be to have runtime dependency on jakarta (which is basic using jaxb-plugins w/ jaxb-plugins-runtime dependency)

But we could give users the right way of having only one XSD / one binding that could be used for javax AND jakarta artifact creation (without depending on backward compatibility).

Creating test project for that use case

**Review by mattrpav — APPROVED** [review]
LGTM

**Pull request merged**